app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt

```diff
@@ -286,11 +286,11 @@ class PasswordCreationActivity : BasePgpActivity(), OpenPgpServiceConnection.OnB
                                         }
                                         .show()
                                 }
-                            } else {
-                                setResult(RESULT_OK, returnIntent)
-                                finish()
                             }
 
+                            setResult(RESULT_OK, returnIntent)
+                            finish()
```

Comment:
This is wrong. With this, you'll once again show the dialog, then immediately finish the activity, making the dialog pointless because it gets dismissed by the finishing activity.

Reply:
but if you put it in the else statement. when oldFile is deleted, APS just waits and doesn't finish the activity.